boards/ti/lp_em_cc2340r5/doc/index.rst

```diff
@@ -0,0 +1,126 @@
++-----------+------------+----------------------+
+| UART      | on-chip    | serial port          |
++-----------+------------+----------------------+
+| SYSTIM    | on-chip    | system timer         |
++------------+------------+---------------------+
+| FLASH     | on-chip    | flash memory         |
++-----------+------------+----------------------+
+
+Other hardware features have not been enabled yet for this board.
+
+Connections and IOs
+===================
+
+All I/O signals are accessible from the BoosterPack connectors. Pin function
+aligns with the LaunchPad standard.
+
++-------+-----------+---------------------+
+| Pin   | Function  | Usage               |
++=======+===========+=====================+
+| DIO0  | GPIO      |                     |
++-------+-----------+---------------------+
+| DIO1  | ANALOG_IO | A4                  |
++-------+-----------+---------------------+
+| DIO2  | ANALOG_IO | A3                  |
++-------+-----------+---------------------+
+| DIO5  | ANALOG_IO | A5                  |
++-------+-----------+---------------------+
+| DIO6  | SPI_CSN   | SPI CS              |
++-------+-----------+---------------------+
+| DIO7  | ANALOG_IO | A0                  |
++-------+-----------+---------------------+
+| DIO8  | GPIO      |                     |
++-------+-----------+---------------------+
+| DIO9  | GPIO      | Button 2            |
++-------+-----------+---------------------+
+| DIO10 | GPIO      | Button 1            |
++-------+-----------+---------------------+
+| DIO11 | SPI_CSN   | SPI CS              |
++-------+-----------+---------------------+
+| DIO12 | SPI_POCI  | SPI POCI            |
++-------+-----------+---------------------+
+| DIO13 | SPI_PICO  | SPI_PICO            |
++-------+-----------+---------------------+
+| DIO14 | GPIO      | Red LED             |
++-------+-----------+---------------------+
+| DIO15 | GPIO      | Green LED           |
++-------+-----------+---------------------+
+| DIO18 | SPI_CLK   | SPI CLK             |
++-------+-----------+---------------------+
+| DIO19 | GPIO      |                     |
++-------+-----------+---------------------+
+| DIO20 | UART0_TX  | UART TX             |
++-------+-----------+---------------------+
+| DIO21 | GPIO      |                     |
++-------+-----------+---------------------+
+| DIO22 | UART0_RX  | UART RX             |
++-------+-----------+---------------------+
+| DIO23 | ANALOG_IO | A8                  |
++-------+-----------+---------------------+
+| DIO24 | ANALOG_IO | A7                  |
++-------+-----------+---------------------+
+
+Programming and Debugging
+*************************
+
+The LP_EM_CC2340R5 requires an external debug probe such as the LP-XDS110 or
+LP-XDS110ET.
+
+Currently there is no debug support in Zephyr for the LP_EM_CC2340R5, and the
+built binaries for this target must be flashed/debugged using either Uniflash or
+Code Composer Studio.
+
```

Comment:
You're right, that's not consistent adding JLink support without mentioning it in this doc. But...

...since we finally removed CRC tool from the PR, I think we should also remove `board.cmake` (which mentions JLink runner). It might be misleading for the user being able to flash a binary that can work **only if** it has been previously built with CCS (otherwise, the system will not boot because of the missing CRC). For the same reason, Uniflash should not be mentioned in this doc.

So, it's probably better adding `board.cmake` in Zephyr (and mentioning Uniflash in the doc) when CRC tool support is added (which is not the case yet). Meanwhile, `board.cmake` should be removed, and Uniflash should not be mentioned in this doc.